cpp/ql/test/experimental/query-tests/Security/CWE/CWE-754/semmle/tests/test.cpp

```diff
@@ -0,0 +1,101 @@
+int scanf(const char *format, ...);
+int globalVal;
+char * globalVala;
+int * globalValp;
+char globalVala2;
+int functionWork1() {
+  int i;
+  char a[10];
+  int b;
+  int *p = &b;
+  if (scanf("%i", &i) != 1) // GOOD
+    return -1;
+  if (scanf("%s", a) != 1) // GOOD
+    return -1;
+  if (scanf("%i", p) != 1) // GOOD
+    return -1;
+  return i;
+}
+
+int functionWork1_() {
+  int i;
+  char a[10];
+  int b;
+  int *p = &b;
+  int r;
+  r = scanf("%i", &i);
+  if (r != 1) // GOOD
+    return -1;
+  r = scanf("%s", a);
+  if (r == 1) // GOOD
+    return -1;
+  r = scanf("%i", p);
+  if (r != 1) // GOOD
+    return -1;
```

Comment:
Technically these are good because `a` and `p` are not being used. Maybe do something like:
```suggestion
  r = scanf("%s", a);
  if (r == 1)
    char a0 = a[0]; // GOOD
  r = scanf("%i", p);
  if (r == 1)
    int pv = *p;  // GOOD
```

Reply:
i added usage via branched return.
if you do not mind?

Reply:
That's fine, as long as there's a use.